### PR TITLE
add sync-delimited set-user-password get-vcpus set-vcpus cmd support

### PR DESCRIPTION
**Pull request by Zhengtong**
depend on https://github.com/avocado-framework/avocado-vt/pull/496
id:1328001 1327995 1327990

### CODE REVIEW

File: qemu/tests/qemu_guest_agent.py [hunk 1]
```diff
@@ -301,6 +301,83 @@ def gagent_check_halt(self, test, params, env):
...
+        vcpus_num = len(vcpus_info)
+        error.context("the vcpu number:%d" % vcpus_num, logging.info)
+        if vcpus_num < 2:
+            raise error.TestNAError("the vpus number of guest should be more"
+                                    " than 1")
+        vcpus_info[vcpus_num - 1]["online"] = False
+        del vcpus_info[vcpus_num - 1]["can-offline"]
+        action = {'vcpus': [vcpus_info[vcpus_num - 1]]}
+        self.gagent.set_vcpus(action)
+
```
Comment (PandaWei) [inline qemu/tests/qemu_guest_agent.py]: I tested your new cases, they all passed.
But you doesn't check the return of get_vcpus/set_vcpus.
If you update it, I'd like to give it ACK.

Comment (Zhengtong) [inline qemu/tests/qemu_guest_agent.py]: Hi  @PandaWei , The return check   is done in function cmd() which is called by get_vcpus/set_vcpus,  I think it is enough to handle the return check. what's your opinion?

Comment (PandaWei) [inline qemu/tests/qemu_guest_agent.py]: Maybe I don't say it clear.
For example,
The output of get_vcpus as following,
[{u'logical-id': 0, u'can-offline': False, u'online': True}, {u'logical-id': 1, u'can-offline': True, u'online': True}]

After we call set_vcpus, We should check if the cpu's attributes have been changed , for example the 'can-offline'.
Is it right?  If i'm wrong,  Please point it out, Thanks in advance.

Comment (Zhengtong) [inline qemu/tests/qemu_guest_agent.py]: I agree with you.  updated the code, please check again  :0

**Comment by PandaWei** [comment]
ACK

File: qemu/tests/qemu_guest_agent.py [hunk 2]
```diff
@@ -301,6 +301,87 @@ def gagent_check_halt(self, test, params, env):
...
+    @error.context_aware
+    def gagent_check_set_vcpus(self, test, params, env):
+        """
+        Execute "guest-set-vcpus" command to guest agent
+       :param test: kvm test object
+       :param params: Dictionary with the test parameters
+       :param env: Dictionary with test environment.
+        """
+        error.context("get the cpu number of the testing guest")
+        vcpus_info = self.gagent.get_vcpus()
```
Comment (CongLi) [inline qemu/tests/qemu_guest_agent.py]: If get_vcpus() return False, then len(vcpus_info) will raise error.

Comment (Zhengtong) [inline qemu/tests/qemu_guest_agent.py]: updated

File: qemu/tests/qemu_guest_agent.py [hunk 3]
```diff
@@ -301,6 +301,95 @@ def gagent_check_halt(self, test, params, env):
...
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        error.context("Check guest agent command 'guest-sync-delimited'",
+                      logging.info)
+        self.gagent.sync("guest-sync-delimited")
+
+    @error.context_aware
+    def _gagent_verify_password(self, vm, new_password, user='root'):
```
Comment (CongLi) [inline qemu/tests/qemu_guest_agent.py]: user is not used.

**Comment by CongLi** [comment]
LGTM,
Acked-by: Cong Li <coli@redhat.com>

**Comment by qunfang** [comment]
(01/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_sync_delimited: PASS (73.09 s)
 (02/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_user_password.no_crypto: PASS (76.27 s)
 (03/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_user_password.crypto: PASS (10.81 s)
 (04/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_get_vcpus: PASS (74.85 s)
 (05/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_vcpus: PASS (9.18 s)
 (06/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_sync_delimited: PASS (77.80 s)
 (07/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_user_password.no_crypto: PASS (64.79 s)
 (08/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_user_password.crypto: PASS (11.87 s)
 (09/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_get_vcpus: PASS (61.13 s)
 (10/12) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_set_vcpus: PASS (9.01 s)
 (11/12) smp_8.8192m.repeat1.post_check.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64.io-github-autotest-qemu.check_coredump: WARN (14.90 s)
 (12/12) smp_8.8192m.repeat1.post_check.Host_RHEL.m7.u2.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.ppc64le.io-github-autotest-qemu.check_coredump: WARN (0.87 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 2 | INTERRUPT 0
JOB HTML   : /root/test_home/qzhang/staf-kvm-devel/workspace/job-results/job-2016-05-23T03.00-aada491/html/results.html
TESTS TIME : 484.57 s

 Acked-by: qzhang@redhat.com

**Comment by CongLi** [comment]
Thanks @Zhengtong @PandaWei @qunfang .

**Pull request merged**